beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -179,6 +191,7 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
                     data_columns.extend(data.clone())
                 }
 
+                *attempt += 1;
```

Comment:
it seems a bit weird incrementing here at first, but it makes sense to me now -  this line is only reached when all requests are in the `Complete` state. might be worth adding a comment here to explain. (because moving this to a few lines earlier would break this, but it isnt immediately obvious to me.